### ISSUE TITLE
Color aesthetic not showing in legend for v2

### ISSUE THREAD

**Issue by wtcooper** [hadley/ggplot2#1473]
After upgrading to ggplot v2, I noticed the color aesthetic no longer shows in the legend guide for some plots, although it shows correctly in the plot.  I tried with two installs of ggplot, and it only fails with v2.  Here's an example:

```R
library(ggplot2)

data <- as.data.frame(USPersonalExpenditure) # data from package datasets
data$Category <- as.character(rownames(USPersonalExpenditure)) # this makes things simpler later
data$val = 1:5

ggplot(data, aes(y = Category, color = val)) +
		labs(x = "Expenditure",y = "Category") +
		geom_segment(aes(x = data$"1940",y = Category,xend = data$"1960",yend = Category),size = 1) 
```

It works as expected with version ggplot2_1.0.1, but fails with version ggplot2_2.0.0.
 
If this is not a bug, how would one get the legend to correctly show the color in v2?

**Comment by hadley** [comment]
Works for me.

Note that you should be doing:

```R
ggplot(data, aes(y = Category, color = val)) +
  labs(x = "Expenditure",y = "Category") +
  geom_segment(aes(x = `1940`,y = Category,xend = `1960`,yend = Category),size = 1) 
```

Otherwise if you ever facet your data it will be effectively randomly assigned across the facets.

**Comment by wtcooper** [comment]
Could be the remote desktop issue as discussed here (http://r.789695.n4.nabble.com/R-base-package-grid-does-not-output-raster-image-td4672188.html), just checked on my local machine and worked OK after I updated to v2.   We recently updated our Server OS about same time as I updated to ggplot v2, and haven't fixed the color rendering issue on new OS yet as described in the post.  I'll see if that fixes the issue to get the old plots worked correctly. 

Thanks
